### ISSUE TITLE
Update numbering and bullet types

### ISSUE THREAD

**Issue by anermina** [metanorma/metanorma-iec#197]
* Numbering types in Foreword's item list differ.
Generated (left) vs. original (right):
![image](https://user-images.githubusercontent.com/55683568/220417475-d3181bc2-4402-4acf-b346-42c86bdf4b4e.png)

* Bullet types in unnumbered item lists are different.
Generated (left) vs. original (right):
![image](https://user-images.githubusercontent.com/55683568/220418550-a591abc8-3c4c-4371-a73e-9d3fa7ebda46.png)

* Numbering types in item lists of the multi-paragraph notes differ.
Generated (left) vs. original (right):
![image](https://user-images.githubusercontent.com/55683568/220419493-5fcefab6-f19b-4e28-9879-42b59c472586.png)

_Related to https://github.com/metanorma/mn-samples-bsi/issues/218._

**Comment by opoudjis** [comment]
Confirmed in random document https://webstore.iec.ch/preview/info_iecfdis60269-1%7Bed5.0%7Den.pdf

**Comment by opoudjis** [comment]
Foreword:

Actually, that foreword is automatically inserted from the Boilerplate—it's the legal statement in the boilerplate; and I've already forced it to be numbered with 1). As I have commented in code,

````
      # Incredibly, the numbered boilerplate list in IEC is NOT A LIST,
      # and it violates numbering conventions for ordered lists
      # (arabic not alpha)
````

And that code has been there for 5 years.

But Nermina has checked PDF not DOC; the document didn't even generate DOC output. So:

Bullet lists:

The correct treatment of bulleted lists in the sample document is hyphen level 1, bullet level 2. In the foreword though they are bullet level 1.

> • reconfirmed,
> • withdrawn, or
> • revised.

 The Word template supports both List Bullet and List Dash. Nothing in https://www.iec.ch/standards-development/writing-and-formatting#iec-styles says which style is used where, so I have no clear sense that dash is to be avoided in favour of bullet, or that there is a principled choice between the two

Nested numbered lists:

In clause 4.5.2, there is a list (alphabetic numbering) and a note within a list item, which itself contains a list; so the list in the note is a second level list. 

The ordering of the Word List Number style is a) 1) i) a) 1) ; the List Number Alt style is 1) a) i)

The numbering implies that notes within numbered lists are being considered as a list level:

a) 1) i) ---

* List a)
  * Note 1)
    * List i) 

But what happens if there is an unnumbered note?

* List a)
  * Note
    * List 1)
?

Treating notes as a numbered list level is frankly oddball, and there is nothing about it in ISO/IEC DIR 2, nor any relevant guidance on the IEC website.
   

This is what is currently happening:

| Format  | Numbering types in Foreword's item list | Bullet types in unnumbered item lists | Numbering types in item lists of the multi-paragraph notes |
| ------------- | ------------- | ------------- | ------------- |
| Presentation XML  |  `//boilerplate/legal-statement/clause/ol[@type = 'alphabet']`   |   |     |
| HTML  |  `<ol type="a" id="_4ba7e7cf-8b86-0df4-6dfa-755dcb1cb123" class="alphabet">`   |  Hyphen |   number  |
| DOC |   `<p>...1)...`  |  Bullet |   number  |
| PDF |  `a)`    |  Bullet  |  number   |

So my current response is:

* I will action the numbered list in the Foreword, it is only fixed in Word, but the fix in PDF and HTML is much more easy.
* I will NOT action bullets vs dashes without guidance from IEC editorial: there is no clear guidance on when to use bullets and when to use dashes, and the IEC documents I am seeing are inconsistent internally.
* I will NOT action List - Note - List numbering without guidance from IEC editorial: the behaviour I am inferring for List - Note - List, that Note counts as a list level number, is odd and undocumented.

**Comment by opoudjis** [comment]
I see that the boilerplate in IEC is still XML not Asciidoc; fixing that.

**Comment by opoudjis** [comment]
@ronaldtse We need to get clarification from IEC on the styling of dashed vs bulleted lists (whether there is any editorial rule, or it is just authorial whim), and on the numbering of ordered lists within numbered notes (examples, etc) within ordered lists.

Or, we leave these as is, and close this ticket, now that the numbering of foreword lists is fixed for PDF and HTML. Your call.

**Comment by ronaldtse** [comment]
@opoudjis IEC has shared private documents and XMLs with us that should already fully address this question.

@anermina can you please help answer @opoudjis with the content here? Thanks[
https://github.com/metanorma/mn-samples-iec-private

**Comment by opoudjis** [comment]
@anermina Please investigate

**Comment by anermina** [comment]
Investigating.

**Comment by anermina** [comment]
| Document | Numbering types in Foreword's item list | Bullet types in unnumbered item lists | Numbering types in numbered item lists | Numbering types in item lists of the multi-paragraph notes | Numbering types in nested numbered lists
| -- | -- | -- | -- | -- | -- |
| 121 proof entries 2020-12-23 | N/A | N/A | N/A | bullet | N/A
| iec-60050-192.sts.nisoiec-60050-192.sts.niso_testAlex | Arabic numbers followed by the right parenthesis - 1), 2),... | bullet  | N/A | N/A | N/A
| IEV 103 snapshot | N/A  | N/A | multiple definitions: Arabic ordinal numbers - 1., 2.,... ![image](https://github.com/metanorma/metanorma-iec/assets/55683568/17f66f6e-1ff5-48d4-8a31-3796d8b40eb2) cases: Arabic numbers between parentheses - (1), (2),... ![image](https://github.com/metanorma/metanorma-iec/assets/55683568/a8726f0f-a32e-4216-8106-9c698bec4365) | bullet | N/A
| IEV 192 snapshot | N/A  | N/A |  multiple definitions: Arabic numbers followed by the right parenthesis - 1), 2),... ![image](https://github.com/metanorma/metanorma-iec/assets/55683568/3b76a4d5-c6a5-49e1-9629-a24aa1dad1cf) | N/A | N/A
| iec62830-8{ed1.0}en | Arabic numbers followed by the right parenthesis - 1), 2),... | bullet in the preface, hyphen in 5.1.2 and 5.1.3 | loweralpha followed by the right parenthesis for the first level, Arabic numbers followed by the right parenthesis for the second level, and hyphens for the third: `The test report shall include at least the following information: a) mandatory: 1) reference to this document; 2) shape, weight and dimensions of the tested supercapacitor; 3) environmental condition for test: – external load; – atmospheric conditions` | N/A | N/A
| iec62899-202-4{ed1.0}en | Arabic numbers followed by the right parenthesis - 1), 2),... | bullet | procedures/steps: loweralpha followed by the right parenthesis: `<label>a)</label>` | N/A | N/A
| iec63159-2-2{ed1.0}en | Arabic numbers followed by the right parenthesis - 1), 2),... | bullet | N/A | N/A | N/A

In conclusion, I believe the numbering type in Foreword's item list should be 1), 2),...; bullet types in unnumbered item lists and multi-paragraph notes should be `bullet`; numbering type in numbered items list depends on the case. Nested numbered lists were unfortunately not found in the sample documents.

**Comment by opoudjis** [comment]
My thanks @anermina !

**Comment by opoudjis** [comment]
I found the List–Note–List convention of treating the note as a list level counterintuitive, and Nermina has not found corroboration, so I'm going to decline to implement it.

No consistency in numbered lists with regard to parentheses. I could implement alphabetic numbering for steps, as has been done in NIST (iec62899-202-4), but alphabetic numbering is used in iec62830-8 anyway. 

Definitions have Arabic numbering, and we do treat them as ordered lists. So at the moment multiple definitions are being rendered with alphabetic numbering, and shouldn't be: that will be particularly visible in IEV. We do support arabic numbering for the foreword; we can do that for definitions as well.

**Comment by opoudjis** [comment]
... As it turns out, the IEC realisation of Forewords, at least when I had access to a Word template (can't find one now) was not a list at all—which is why it could have different numbering. I'm doing that rendering automatically in Word for definitions as well.
